File: src/api/accessibility/multi_criteria/MultiCriteriaRequest.cs
using System;
using System.Collections.Generic;
using NetTopologySuite.Geometries;
using DVAN.Population;
using DVAN.Routing;

namespace DVAN.API
{
    /// <summary>
    /// Multi-criteria Request.
    /// </summary>
    public class MultiCriteriaRequest
    {
        /// <summary>
        /// infrastructure parameters.
        /// </summary>
        public Dictionary<string, InfrastructureParams> infrastructures { get; set; }

        public PopulationRequestParams population { get; set; }

        public RoutingRequestParams? routing { get; set; }
    }

    /// <summary>
    /// Infrastructure parameter.
    /// </summary>
    public class InfrastructureParams
    {
        /// <summary>
        /// Weight of infrastructure in multi-criteria..
        /// </summary>
        /// <example>0.8</example>
        public double infrastructure_weight { get; set; }

        /// <summary>
        /// Ranges (in sec) to be used by calculation.
        /// </summary>
        /// <example>[180, 360, 540, 720, 900]</example>
        public List<double> ranges { get; set; }

        /// <summary>
        /// Distance-Decay for every catchment (ranges).
        /// </summary>
        /// <example>[1.0, 0.7, 0.5, 0.3, 0.1]</example>
        public List<double> range_factors { get; set; }

        /// <summary>
        /// Facility Locations in geographic coordinates.
        /// </summary>
        /// <example>[[9.11, 51.23], [9.35, 50.98], [10.02, 52.10]]</example>
        public double[][] facility_locations { get; set; }
    }
}

File: src/main/java/org/tud/oas/api/accessibility/multi_criteria/MultiCriteriaRequest.java
package org.tud.oas.api.accessibility.multi_criteria;

import java.util.List;
import java.util.Map;

import org.tud.oas.requests.AccessResponseParams;
import org.tud.oas.requests.DecayRequestParams;
import org.tud.oas.requests.DemandRequestParams;
import org.tud.oas.requests.RoutingRequestParams;
import org.tud.oas.requests.SupplyRequestParams;

import com.fasterxml.jackson.annotation.JsonProperty;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(name = "MultiCriteriaRequest", description = """
                Multi-criteria Request.
                """)
public class MultiCriteriaRequest {
        @Schema(name = "infrastructures", description = """
                        infrastructure parameters.
                        """)
        public Map<String, InfrastructureParams> infrastructures;

        public DemandRequestParams demand;

        public RoutingRequestParams routing;

        @Schema(name = "return_all", description = """
                        If true all individuel results will be returned.
                        Else only multi-criteria result.
                        """)
        public Boolean return_all;

        @Schema(name = "return_weighted", description = """
                        If true all access weighted by demand-count will be computed ("{name}_weighted").
                        """)
        public Boolean return_weighted;

        @JsonProperty("response")
        public AccessResponseParams response_params;
}

@Schema(name = "InfrastructureParams", description = """
                Infrastructure parameter.
                """)
class InfrastructureParams {
        @Schema(name = "infrastructure_weight", description = """
                        Weight of infrastructure in multi-criteria..
                        """, example = "0.8")
        public float infrastructure_weight;

        public DecayRequestParams decay;

        public SupplyRequestParams supply;
}
